fusor-ember-cli/app/controllers/subscriptions/credentials.js

```diff
@@ -53,7 +57,62 @@ export default Ember.Controller.extend({
     } else {
       return 'loginPortal';
     }
-  }.property('model.isAuthenticated')
+  }.property('model.isAuthenticated'),
+
+  isDisconnected: Ember.computed.alias('controllers.deployment.model.is_disconnected'),
+  hasManifestFile: Ember.computed.notEmpty('manifestFile'),
+  noManifestFile: Ember.computed.empty('manifestFile'),
+
+  contentProviderType: function() {
+    return (this.get('isDisconnected') ? "disconnected" : "redhat_cdn");
+  }.property('isDisconnected'),
+
+  contentProviderTitle: function() {
+    return (this.get('isDisconnected') ? "Disconnected" : "Red Hat CDN");
+  }.property('isDisconnected'),
+
+  isDisconnectedSelected: function() {
+    return (this.get('contentProviderType') === 'disconnected');
+  }.property('contentProviderType'),
+
+  actions: {
+    providerTypeChanged: function() {
+      return this.set('isDisconnected', this.get('isDisconnectedSelected'));
+    },
+
+    uploadManifest: function() {
+      var self = this;
+      var manifestFile = document.getElementById('manifest-file-field').files[0];
+      var formData = new FormData();
+      formData.append('manifest_file[name]', manifestFile.name);
+      formData.append('manifest_file[file]', manifestFile);
+      formData.append('manifest_file[deployment_id]', this.get('deploymentId'));
+
+      console.log('action: uploadManifest, POST /fusor/api/v21/subscriptions/upload');
+      //ic-ajax request
+      request({
+        type: 'POST',
+        url: '/fusor/api/v21/subscriptions/upload',
```

Comment:
@jmrodri, I'm using this endpoint that you mentioned in IRC and using POST. I can change to PUT if you want.